File: PPC/Assignments/KnapsackProblem2/KnapsackProblemGo/knapsack/channel/messages.go
package channel

import "KnapsackProblemGo/knapsack"

type ProcessIndividuals struct {
	idx        int
	population []*knapsack.Individual
}

type ProcessedIndividuals struct {
	idx        int
	population []*knapsack.Individual
}

type ProcessCrossoverIndividuals struct {
	idx        int
	chunkSize  int
	population []*knapsack.Individual
}

type ProcessedCrossoverIndividuals struct {
	idx           int
	chunkSize     int
	newPopulation []*knapsack.Individual
}

File: PPC/Assignments/KnapsackProblem2/KnapsackProblemKotlin/src/main/kotlin/knapsack/channel/Messages.kt
package knapsack.channel

import Individual

sealed interface Messages {
    data class ProcessIndividuals(
        val idx: Int,
        val population: Array<Individual>
    ) : Messages

    data class ProcessedIndividuals(
        val idx: Int,
        val population: Array<Individual>
    ) : Messages

    data class ProcessCrossoverIndividuals(
        val idx: Int,
        val chunkSize: Int,
        val population: Array<Individual>,
    ) : Messages

    data class ProcessedCrossoverIndividuals(
        val idx: Int,
        val chunkSize: Int,
        val newPopulation: Array<Individual>,
    ) : Messages
}
